src/components/DeliverableView/DeliverableStatusBadge.tsx

```diff
@@ -0,0 +1,90 @@
+import { useTheme } from '@mui/material';
+import { Badge } from '@terraware/web-components';
+import { BadgeProps } from '@terraware/web-components/components/Badge';
+import { useMemo } from 'react';
+import strings from 'src/strings';
+import { DeliverableStatusType } from 'src/types/Deliverables';
+
+type DeliverableStatusBadgeProps = {
+  status: DeliverableStatusType;
+};
+
+const DeliverableStatusBadge = (props: DeliverableStatusBadgeProps): JSX.Element => {
+  const { status } = props;
+  const theme = useTheme();
+
+  const getBadgePropsForStatus = useMemo(
+    () => (): BadgeProps => {
+      let badgeProps: BadgeProps;
+
+      switch (status) {
+        case 'Approved':
+          badgeProps = {
```

Comment:
You can just return the object from each of these and save the let / assignment.

Reply:
This was an intentional decision so there would only be one `return` statement in the function, but I updated it since this change reduces the lines of code and is more direct.